src/bootstrap/setup.rs

```diff
@@ -89,6 +94,12 @@ pub fn setup(src_path: &Path, profile: Profile) {
 
     println!();
 
+    #[cfg(not(target_os = "linux"))]
+    {
+        t!(llvm_config(&path));
+        println!();
+    }
```

Comment:
What is this doing? It looks like it will panic on all platforms other than linux which is definitely not the right behavior.

Reply:
Oh wait I missed

> I just need to write the llvm-config detection logic for other platforms

I would use `Command::new("llvm-config").args("--version").output.unwrap_or(false, |output| output.status.success())` or something like that.

Reply:
That said, on Ubuntu at least llvm-config will normally have a version suffix (like `llvm-config-8`) so this isn't super reliable.

Reply:
Another problem with this approach is that we need to get the absolute path to `llvm-config`, not just check if it's there. At least on linux, I'm getting an error when using the following config.toml file:

```
[target.x86_64-unknown-linux-gnu]
llvm-config = "llvm-config"
```

error:

```
failed to execute command: "/home/alarsyo/work/rust/llvm-config" "--bindir"
error: No such file or directory (os error 2)
```

Reply:
I'll write some code to walk through the `PATH` directories and search for an `llvm-config` executable, though I don't have a windows machine to test the windows-specific issues that could arise (`.exe` suffix, possibly different `PATH` syntax..)

Reply:
@Mark-Simulacrum this sounds pretty much like rewriting the `which` crate, are you sure it's not worth pulling in the dependency?

Reply:
I don't think we need to be as fancy as the which crate here, which does "is it executable" and such checking. I would expect PATH differences to be dealt with via split_paths. I expect this to be 3-4 lines of code -- if it ends up more than that we can pull in which.